In [99]:
#import needed modules
import pandas as pd
import numpy as np
import sklearn

In [100]:
col_names= [ 'patient', 'Age','Partner_Score','Self_Score','P_S','victim','PH_Score','PY_Score','E_Score','S_Score']

In [101]:
#importing first model-survey results
m1 = pd.read_csv("model - m1.csv", header= 0, names= col_names)

#iporting 2nd model-survey results
#m2 = pd.read_csv("model - m2.csv", header= 0, names= col_names)


In [102]:
df = pd.DataFrame(m1.data, columns=col_names) # load the dataset as a pandas data frame

AttributeError: 'DataFrame' object has no attribute 'data'

In [103]:
vict=pd.concat([m1,m2])

In [104]:
#verify merge
vict.head()

patient  Age  Partner_Score  Self_Score  P_S victim  PH_Score  PY_Score  \
0        1   35              2           2    4    Yes         5         2   
1        2   38              0           5    5    Yes         1         6   
2        3   20              1           1    2    Yes         6         0   
3        4   16              2           5    7    Yes         2         5   
4        5   34              5           2    7    Yes         3         4   

   E_Score  S_Score  
0        2        0  
1        4        5  
2        0        4  
3        4        4  
4        6        6

In [105]:
for Unknown in vict1.victim:
    print(Unknown)

Yes
Yes
Yes
Yes
Yes
Yes
Unknown
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Unknown
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
No
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Unknown
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Unknown
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Yes
Unknown
No
Unknown
Unknown
Yes
Unknown
No
Yes
Unknown
Yes
Unknown
Yes
Unknown
Unknown
Yes
Yes
Yes
No
Unknown
No
No
Yes
Yes
Yes
Unknown
Unknown
Yes
Unknown
Yes
Yes
Unknown
No
Yes
Yes
Unknown
Yes
Unknown
Yes
Yes
Yes
Yes
Unknown
Unknown
Yes
Yes
Unknown
Yes
Unknown
Yes
Yes
Yes
Yes
Yes
Yes
No
Yes
Unknown
Yes
Yes
Unknown
Unknown
Yes
Yes
Yes
Yes
Yes
No
Unknown
Unknown
Unknown
Yes
Yes
Unknown
Yes
Unknown
Yes
Yes
Unknown
Yes
Unknown
Unknown
Unknown
Unknown
Unknown
No
Yes
Unknown
No
Unknown
Unknown
Yes
Unknown
Unknown
Yes
Yes
Unknown
Yes
Yes
Unknown


In [106]:
#drop last 4 columns in new df
#create new df to only identify victim = yes / no/unknown
vict1= vict.drop(columns=['patient','PH_Score','PY_Score','E_Score','S_Score'], axis=0)

In [107]:
#verify colunms succefully dropped
vict1.head()

Age  Partner_Score  Self_Score  P_S victim
0   35              2           2    4    Yes
1   38              0           5    5    Yes
2   20              1           1    2    Yes
3   16              2           5    7    Yes
4   34              5           2    7    Yes

In [108]:
vict1.iloc[:50]

Age  Partner_Score  Self_Score  P_S   victim
0    35              2           2    4      Yes
1    38              0           5    5      Yes
2    20              1           1    2      Yes
3    16              2           5    7      Yes
4    34              5           2    7      Yes
5    28              3           1    4      Yes
6    35              1           0    1  Unknown
7    23              2           4    6      Yes
8    28              4           0    4      Yes
9    31              0           5    5      Yes
10   32              1           5    6      Yes
11   34              5           4    9      Yes
12   45              0           2    2      Yes
13   39              3           4    7      Yes
14   40              0           3    3      Yes
15   38              0           3    3      Yes
16   34              2           4    6      Yes
17   18              5           4    9      Yes
18   28              3           0    3      Yes
19   44              1           5    6      Yes
20   27              3           4    7      Yes
21   20              0           5    5      Yes
22   28              5           0    5      Yes
23   35              0           1    1  Unknown
24   40              3           2    5      Yes
25   32              5           3    8      Yes
26   46              4           3    7      Yes
27   44              1           3    4      Yes
28   43              2           4    6      Yes
29   38              4           0    4      Yes
30   27              5           1    6      Yes
31   38              5           1    6      Yes
32   42              2           0    2      Yes
33   17              5           5   10      Yes
34   50              0           2    2      Yes
35   44              4           5    9      Yes
36   37              0           4    4      Yes
37   37              2           4    6      Yes
38   34              5           3    8      Yes
39   17              1           5    6      Yes
40   42              0           2    2      Yes
41   30              2           5    7      Yes
42   16              2           5    7      Yes
43   24              0           0    0       No
44   24              1           5    6      Yes
45   36              5           1    6      Yes
46   28              2           1    3      Yes
47   23              2           1    3      Yes
48   20              3           1    4      Yes
49   47              4           1    5      Yes

In [109]:
#convert victim dtype to category from object in order to encode
vict1["victim"] = vict1["victim"].astype('category')

In [110]:
#recode victim column
#label encoder column victim
vict1["victim_cat"] = vict1["victim"].cat.codes

In [111]:
#verify successful encoding
vict1.head()
#drop column victim to have all int64 dtype
vict1= vict1.drop(columns=['victim'])

In [112]:
#verify victim column was successfully dropped
#verify dtype int64 for all
vict1.head()

Age  Partner_Score  Self_Score  P_S  victim_cat
0   35              2           2    4           2
1   38              0           5    5           2
2   20              1           1    2           2
3   16              2           5    7           2
4   34              5           2    7           2

In [149]:
label_names=['Yes','No','Unknown']

In [150]:
feature_names=['Age','Partner_Score','Self_Score','P_S']

In [160]:
# Organize data
label_names = label_names
labels = vict1.iloc[:, 3].values
feature_names = feature_names
features = vict1.iloc[:, -1].values

In [161]:
# Look at our data
print(label_names)
print(labels[0])
print(feature_names[0])
print(features[0])

['Yes', 'No', 'Unknown']
4
Age
2


In [118]:
vict1['victim_cat'].head()

0    2
1    2
2    2
3    2
4    2
Name: victim_cat, dtype: int8

In [119]:
vict1['victim_cat'].tail()

95    2
96    1
97    2
98    2
99    1
Name: victim_cat, dtype: int8

In [120]:
#print shape of new data
print(vict1.shape)

(200, 5)


In [123]:
#verify if data is balanced
print(vict1.groupby('victim_cat').size())

victim_cat
0     11
1     44
2    145
dtype: int64


### Victim identified = No (11)
### Victime identified= Unknown (44)
### Victim identified= Yes (145)

In [124]:
print(vict.groupby('Age').size())

Age
14    3
15    2
16    9
17    6
18    3
19    4
20    6
21    3
22    5
23    7
24    5
25    3
26    2
27    5
28    9
29    5
30    5
31    7
32    7
33    7
34    7
35    5
36    3
37    5
38    8
39    8
40    6
41    7
42    6
43    4
44    8
45    6
46    6
47    5
48    4
49    4
50    5
dtype: int64


In [162]:
vict1.head()

Age  Partner_Score  Self_Score  P_S  victim_cat
0   35              2           2    4           2
1   38              0           5    5           2
2   20              1           1    2           2
3   16              2           5    7           2
4   34              5           2    7           2

In [164]:
vict1.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 200 entries, 0 to 99
Data columns (total 5 columns):
Age              200 non-null int64
Partner_Score    200 non-null int64
Self_Score       200 non-null int64
P_S              200 non-null int64
victim_cat       200 non-null int8
dtypes: int64(4), int8(1)
memory usage: 13.0 KB


In [166]:
vict1.describe()

Age  Partner_Score  Self_Score         P_S  victim_cat
count  200.000000      200.00000  200.000000  200.000000  200.000000
mean    32.735000        1.32000    1.315000    2.635000    1.670000
std     10.215998        2.72299    3.063117    4.309832    0.576857
min     14.000000       -5.00000   -5.000000   -9.000000    0.000000
25%     24.000000        0.00000   -1.000000    0.000000    1.000000
50%     33.000000        2.00000    2.000000    3.000000    2.000000
75%     41.000000        3.25000    4.000000    6.000000    2.000000
max     50.000000        5.00000    5.000000   10.000000    2.000000

In [174]:
#assign features to variable x
X = vict1.iloc[:, -1].values 

In [175]:
print(X)

[2 2 2 2 2 2 1 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 1 2 2 2 2 2 2 2 2 2 2 2 2 2
 2 2 2 2 2 2 0 2 2 2 2 2 2 2 2 2 2 2 2 2 2 1 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2
 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 2 1 2 2 2 2 2 2 2 2 2 2 1 0 1 1 2 1 0 2 1 2
 1 2 1 1 2 2 2 0 1 0 0 2 2 2 1 1 2 1 2 2 1 0 2 2 1 2 1 2 2 2 2 1 1 2 2 1 2
 1 2 2 2 2 2 2 0 2 1 2 2 1 1 2 2 2 2 2 0 1 1 1 2 2 1 2 1 2 2 1 2 1 1 1 1 1
 0 2 1 0 1 1 2 1 1 2 2 1 2 2 1]


In [176]:
#assign target variable
y = vict1.iloc[:, 3].values 

In [177]:
print(y)

[ 4  5  2  7  7  4  1  6  4  5  6  9  2  7  3  3  6  9  3  6  7  5  5  1
  5  8  7  4  6  4  6  6  2 10  2  9  4  6  8  6  2  7  7  0  6  6  3  3
  4  5  4  4  3  6  7  7  2  2  1  6  7  2  6 10  7  3  3  8  4  5  4  4
  5  4  7  2  8  4  7  5  5  5  6  9  8  5  6  7  4  7  1  5  8  4 10  5
  8  3  3  6  7 -7  0 -4 -2  6 -3  0 10 -1  8 -1  1 -2 -1  3  7  1  0 -9
  0  0  4  4  7 -3 -9  2 -5  4  5 -5  0  1  3 -4  1 -6  3  2  3  7 -4 -1
  5  3 -4  2 -2  2  6  7  1  5  3  0  6 -7  1  2 -9 -7  7  1  5  2  3  0
 -5 -4 -5  5  1 -4  2 -5  2  2 -5  6 -5 -1 -8 -2 -3  0  1 -3  0 -6 -3  7
 -6 -5  1  2 -6  5  2 -2]


In [179]:
np.reshape(y)

TypeError: reshape() missing 1 required positional argument: 'newshape'

In [178]:

# importing necessary libraries 

from sklearn import datasets 

from sklearn.metrics import confusion_matrix 

from sklearn.model_selection import train_test_split 

  

  

X_train, X_test, y_train, y_test = train_test_split(X, y, random_state = 0) 

  
# training a DescisionTreeClassifier 

from sklearn.tree import DecisionTreeClassifier 

dtree_model = DecisionTreeClassifier(max_depth = 2).fit(X_train, y_train) 

dtree_predictions = dtree_model.predict(X_test) 

  
# creating a confusion matrix 


ValueError: Expected 2D array, got 1D array instead:
array=[2. 2. 1. 2. 2. 2. 2. 1. 2. 2. 1. 2. 1. 2. 2. 2. 2. 2. 2. 2. 2. 1. 1. 1.
 2. 2. 1. 2. 2. 2. 1. 2. 2. 2. 2. 2. 2. 2. 2. 2. 0. 2. 2. 2. 1. 1. 1. 1.
 2. 2. 2. 2. 2. 2. 2. 2. 2. 2. 0. 2. 2. 2. 2. 2. 2. 1. 1. 2. 1. 2. 2. 2.
 1. 2. 2. 2. 2. 2. 2. 2. 1. 2. 2. 0. 0. 2. 2. 2. 1. 2. 1. 0. 2. 2. 2. 2.
 0. 2. 2. 2. 0. 1. 2. 2. 2. 2. 1. 1. 2. 2. 1. 2. 2. 2. 2. 2. 2. 1. 1. 2.
 2. 2. 2. 2. 2. 1. 1. 2. 2. 2. 2. 2. 0. 2. 1. 2. 1. 2. 2. 2. 2. 2. 2. 2.
 1. 2. 1. 2. 2. 2.].
Reshape your data either using array.reshape(-1, 1) if your data has a single feature or array.reshape(1, -1) if it contains a single sample.

In [144]:

from sklearn.model_selection import KFold # import KFold

kf = KFold(n_splits=2) # Define the split - into 2 folds 
kf.get_n_splits(X) # returns the number of splitting iterations in the cross-validator

print(kf) 
KFold(n_splits=2, random_state=None, shuffle=False)

KFold(n_splits=2, random_state=None, shuffle=False)


KFold(n_splits=2, random_state=None, shuffle=False)

In [143]:
from sklearn.model_selection import KFold 
#kf = RepeatedKFold(n_splits=5, n_repeats=10, random_state=None) 

for train_index, test_index in kf.split(X):
      print("Train:", train_index, "Validation:",test_index)
      X_train, X_test = X[train_index], X[test_index] 
      y_train, y_test = y[train_index], y[test_index]

Train: [2 3] Validation: [0 1]
Train: [0 1] Validation: [2 3]
